src/transport/grpc/store.rs

```diff
@@ -0,0 +1,345 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//! # Astarte GRPC Store Module
+//!
+//! This module provides an implementation of the PropertyStore.
+//! It defines the `GrpcStore` struct, which handles device properties differently than server
+//! properties, to allow retrieving all device properties directly from the message hub server.
+
+use std::sync::Arc;
+
+use astarte_message_hub_proto::tonic;
+use tokio::sync::Mutex;
+
+use crate::{
+    retention::Missing,
+    store::{
+        OptStoredProp, PropertyInterface, PropertyMapping, PropertyStore, StoreCapabilities,
+        StoredProp,
+    },
+    AstarteType,
+};
+
+use super::{
+    convert::{self, MessageHubProtoError},
+    MsgHubClient,
+};
+
+// Store implementation designed specifically for the grpc connection
+/// Used mainly to request device owned properties to the message hub instead of looking them up in the local storage
+#[derive(Debug, Clone)]
+pub struct GrpcStore {
+    pub(crate) client: Arc<Mutex<MsgHubClient>>,
+}
+
+impl GrpcStore {
+    pub(crate) fn new(client: MsgHubClient) -> Self {
+        Self {
+            client: Arc::new(Mutex::new(client)),
+        }
+    }
+
+    async fn load_properties(
+        &self,
+        ownership: Option<astarte_message_hub_proto::Ownership>,
+    ) -> Result<Vec<StoredProp>, GrpcStoreError> {
+        self.client
+            .lock()
+            .await
+            .get_all_properties(astarte_message_hub_proto::StoredPropertiesFilter {
+                ownership: ownership.map(|o| o.into()),
+            })
+            .await
+            .map(tonic::Response::into_inner)
+            .map_err(GrpcStoreError::from)
+            .and_then(|p| Ok(convert::map_set_stored_properties(p)?))
+    }
+}
+
+/// Error returned while operating on the store of a grpc connection
+/// This store needs to request properties from the message hub server
+/// and has additional errors consequently
+#[non_exhaustive]
+#[derive(thiserror::Error, Debug)]
+pub enum GrpcStoreError {
+    /// Error while retrieving data from the message hub server
+    #[error("Error while retrieving data from the message hub server: {0}")]
+    Status(tonic::Status),
+    /// Error while converting a proto received value to an internal type
+    #[error("Error while converting a proto received value to an internal type: {0}")]
+    Conversion(#[from] MessageHubProtoError),
+}
+
+impl From<tonic::Status> for GrpcStoreError {
+    fn from(value: tonic::Status) -> Self {
+        Self::Status(value)
+    }
+}
+
+impl StoreCapabilities for GrpcStore {
+    type Retention = Missing;
+
+    fn get_retention(&self) -> Option<&Self::Retention> {
+        None
+    }
+}
+
+/// We implement the PropertyStore to override the behavior when retrieving or storing
+/// device owned properies. We only want to store and load from the store server owned properties.
```

Comment:
Nit: Change to `properties`
```suggestion
/// device owned properties. We only want to store and load from the store server owned properties.
```